kubeflow/Makefile

```diff
@@ -51,6 +52,67 @@ apply-cnrm:
 	kustomize build --load-restrictor LoadRestrictionsNone -o $(BUILD_DIR)/gcp_config ./common/cnrm
 	kubectl --context=$(MGMTCTXT) apply -f ./$(BUILD_DIR)/gcp_config
 
+.PHONY: apply-component
+apply-component: validate-values check-name 
+ifeq ($(component_path),)
+	$(error component_path is not set for apply-component rule); 
+endif
+	echo 'Component path: $(component_path)' 
+	$(eval makefile_path = ./$(component_path)/Makefile)
+	echo 'Apply component resources: $(component_path)'
+	@if [ -f $(makefile_path) ] ; then \
+		echo 'Found Makefile, call `make apply` of this component Makefile.'; \
+		NAME=$(NAME) KFCTXT=$(KFCTXT) LOCATION=$(LOCATION) PROJECT=$(PROJECT) $(MAKE) -C $(component_path) apply build_dir=$(BUILD_DIR); \
+	else \
+		echo 'Makefile not found, use kustomize and kubectl to apply resources.'; \
+		component_build_dir=./$(component_path)/$(BUILD_DIR); \
+		rm -rf $$component_build_dir && mkdir -p $$component_build_dir; \
+		kustomize build --load-restrictor LoadRestrictionsNone -o $$component_build_dir ./$(component_path); \
+		kubectl --context=$(KFCTXT) apply -f $$component_build_dir; \
+	fi
```

Comment:
Makefile rules are usually deterministic, they accept no value and produce stable results.
Consider converting this rule to a bash script? I think that simplifies this rule and also make it easier to understand what's going on.
(You may put individual bash scripts at `hack` folder)

Reply:
That makes sense, I had a hard time writing bash script in make rule too. Extracted it to a hack/apply_component.sh.